adoption-metrics/implementation.md

```diff
@@ -0,0 +1,157 @@
+## Instrumenting
+
+### Prometheus
+
+(Docs available [here](https://github.com/prometheus/client_python))
+
+We will reuse the
+[`/metrics`](https://github.com/packit/packit-service/blob/main/packit_service/service/app.py#L52)
+endpoint.
+
+From what I understand, the new measured values will need to be added to
+[Pushgateway](https://github.com/packit/packit-service/blob/main/packit_service/worker/monitoring.py#L29).
+Then use/reuse the metrics definitions in the code ->
+[how to](https://github.com/prometheus/client_python#instrumenting).
+If needed, you can define your
+[own collectors](https://github.com/prometheus/client_python#custom-collectors) but the basic
+ones should be enough.
+
+---
+
+## Gathering data and displaying them into graphs
+
+The data will be gathered from the `/metrics` endpoint.
+
+To aggregate the data dynamically in real time, we will use
+[promql queries](https://prometheus.io/docs/prometheus/latest/querying/basics/).
+
+We have a few options here, each has some pros and cons which are described in each point.
+
+### Grafana
+
+Probably the easiest way to go since grafana provides out-of-the-box support for prometheus.
+We have also some experience with duo prometheus && grafana, so it should be easier for someone
+whom run into a problem with this (he/she can ask someone (=^ ◡ ^=)).
+
+Also since we already have a grafana dashboard listening on `/metrics` endpoint, it should be
+fairly easy to visualize gathered data - just creating a new panel and navigate in the GUI
+to edit panel and providing a reasonable
+[promql queries](https://prometheus.io/docs/prometheus/latest/querying/basics/).
+
+[grafana docs](https://grafana.com/docs/grafana/next/getting-started/get-started-grafana-prometheus/)
+
+Pros:
+
+- out-of-the-box support
+- the easiest solution
+- nice fancy dynamic graphs
+- creating / deleting / editing graphs is easy and well described in the
+  [docs](https://grafana.com/docs/grafana/next/getting-started/get-started-grafana-prometheus/)
+- these graphs should be dynamic (when we want to see graphs in a given period - we can
+  dynamically change this period)
+
+Cons:
+
+- visibility: only internal folks will be able to see the graphs (since our current dashboard
```

Comment:
* If we can do it publicly, why not do that publically.
* Public service can be easier to access also for internal people.

Reply:
Yes, the point is that it would be great to have this data publicly available (for the reasons Franta mentioned above), but it's not necessary (from the key functionality point of view). So there is an option to implement key functionality and later make this data publicly available (using the steps outlined in the PR)

Reply:
Exactly. Let's start with the private Grafana and make a public view as a second step.